pkg/controller/v1alpha2/experiment/manifest/generator.go

```diff
@@ -0,0 +1,158 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultMetricsCollectorTemplateName = "defaultMetricsCollectorTemplate.yaml"
+)
+
+// Generator is the type for manifests Generator.
+type Generator interface {
+	GetRunSpec(e *experimentsv1alpha2.Experiment, experiment, trial, namespace string) (string, error)
+	GetRunSpecWithHyperParameters(e *experimentsv1alpha2.Experiment, experiment, trial, namespace string, hps []*apiv1alpha2.ParameterAssignment) (string, error)
+	GetMetricsCollectorManifest(experimentName string, trialName string, jobKind string, namespace string, metricNames []string, mcs *experimentsv1alpha2.MetricsCollectorSpec) (*bytes.Buffer, error)
+}
+
+// DefaultGenerator is the default implementation of Generator.
+type DefaultGenerator struct {
+	client katibclient.Client
+}
+
+// New creates a new Generator.
+func New() (Generator, error) {
+	katibClient, err := katibclient.NewClient(client.Options{})
+	if err != nil {
+		return nil, err
+	}
+	return &DefaultGenerator{
+		client: katibClient,
+	}, nil
+}
+
+func (g *DefaultGenerator) GetMetricsCollectorManifest(experimentName string, trialName string, jobKind string, namespace string, metricNames []string, mcs *experimentsv1alpha2.MetricsCollectorSpec) (*bytes.Buffer, error) {
+	var mtp *template.Template = nil
+	var err error
+	tmpValues := map[string]string{
+		"Experiment":     experimentName,
+		"Trial":          trialName,
+		"JobKind":        jobKind,
+		"NameSpace":      namespace,
+		"ManagerService": commonv1alpha2.GetManagerAddr(),
+		"MetricNames":    strings.Join(metricNames, ";"),
+	}
+	if mcs != nil && mcs.GoTemplate.RawTemplate != "" {
+		mtp, err = template.New("MetricsCollector").Parse(mcs.GoTemplate.RawTemplate)
+	} else {
+		mctp := defaultMetricsCollectorTemplateName
+		if mcs != nil && mcs.GoTemplate.TemplateSpec != nil {
+			mctp = mcs.GoTemplate.TemplateSpec.TemplatePath
+		}
+		mtl, err := g.client.GetMetricsCollectorTemplates()
+		if err != nil {
+			return nil, err
+		}
+		if mt, ok := mtl[mctp]; !ok {
+			return nil, fmt.Errorf("No MetricsCollector template name %s", mctp)
+		} else {
+			mtp, err = template.New("MetricsCollector").Parse(mt)
+		}
+	}
+	if err != nil {
+		return nil, err
+	}
+	var b bytes.Buffer
+	err = mtp.Execute(&b, tmpValues)
+	if err != nil {
+		return nil, err
+	}
+	return &b, nil
+}
+
+// GetRunSpec get the specification for trial.
+func (g *DefaultGenerator) GetRunSpec(e *experimentsv1alpha2.Experiment, experiment, trial, namespace string) (string, error) {
+	params := trialTemplateParams{
+		Experiment: experiment,
+		Trial:      trial,
+		NameSpace:  namespace,
+	}
+	return g.getRunSpec(e, params)
+}
+
+// GetRunSpecWithHyperParameters get the specification for trial with hyperparameters.
+func (g *DefaultGenerator) GetRunSpecWithHyperParameters(e *experimentsv1alpha2.Experiment, experiment, trial, namespace string, hps []*apiv1alpha2.ParameterAssignment) (string, error) {
```

Comment:
I think namespace and experiment param can be removed since we can get them by `e *experimentsv1alpha2.Experiment`